lua/nvim-tree/utils.lua

```diff
@@ -307,25 +307,57 @@ function M.key_by(tbl, key)
   return keyed
 end
 
----Execute callback timeout ms after the lastest invocation with context. Waiting invocations for that context will be discarded. Caller should this ensure that callback performs the same or functionally equivalent actions.
+local function timer_stop_close(timer)
+  if timer:is_active() then
+    timer:stop()
+  end
+  if not timer:is_closing() then
+    timer:close()
```

Comment:
Manual reference count handling was not necessary; libuv will clear the last reference on handle close.